File: Data Structures & Algorithms/level-order-traversal-of-binary-tree/submission-1.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    private List<List<Integer>> soln = new ArrayList<>();
    public List<List<Integer>> levelOrder(TreeNode root) {
        levelRecursive(root, 0);
        return soln;
    }

    private void levelRecursive(TreeNode curr, int depth)   {
        if (curr == null)   {
            return;
        }

        if (soln.size() == depth)   {
            soln.add(new ArrayList<>());
        }

        soln.get(depth).add(curr.val);
        levelRecursive(curr.left, depth+1);
        levelRecursive(curr.right, depth+1);
    }
}

File: Data Structures & Algorithms/count-good-nodes-in-binary-tree/submission-1.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def goodNodes(self, root: TreeNode) -> int:
        self.count = 0
        self.dfs(root, root.val)
        return self.count

    def dfs(self, node, maxVal):
        if not node:
            return 0
        
        if node.val >= maxVal:
            self.count += 1
            maxVal = node.val

        return 1 + self.dfs(node.left, maxVal) + self.dfs(node.right, maxVal)